src/CppUTestExt/IEEE754ExceptionsPlugin.cpp

```diff
@@ -34,6 +34,53 @@ extern "C" {
     #include <fenv.h>
 }
 
+#if CPPUTEST_FENV_LACKS_SIGNALLING
+
+/*
+ * Missing implementations for Apple, MinGW and Watcom C.
+ */
+
+#if defined(__APPLE__)
+#define CONTROL_WORD __control
+#elif defined(__MINGW32__)
+#define CONTROL_WORD __control_word
+#elif defined(__WATCOMC__)
+#define CONTROL_WORD control_word
+#endif
+
+int feenableexcept (int excepts)
+{
+    fenv_t fenv;
+    unsigned short old_excepts;
+    unsigned short new_excepts = (unsigned short)excepts & FE_ALL_EXCEPT;
+
+    if (fegetenv (&fenv)) return -1;
+
+    old_excepts = fenv.CONTROL_WORD & FE_ALL_EXCEPT;
+    /* Mingw GCC promotes ~<unsigned short> to <int> and then complains about
+     * loss of information when casting from <int> to <unsigned short>. Yikes.
```

Comment:
Exactly. You want to submit the bug? I find their system too cumbersome. Minimal example:
```
#pragma GCC diagnostic error "-Wconversion"
int main(int, char**) {
    unsigned short u = 0x000f;
    u = ~u;
    return (int) u;
}

/* Compiler output (MinGW 4.8.1, 4.9.x, 5.3.0):

OneTimeThrowAwayTests.cpp: In function 'int main(int, char**)':
OneTimeThrowAwayTests.cpp:5:7: error: conversion to 'short unsigned int' from 'int' may alter its value [-Werror=conversion]
     u = ~u;
       ^
*/
```
e.g. no cast should be necessary in the line below.